contracts/NFTMarket.sol

```diff
@@ -450,6 +469,11 @@ contract NFTMarket is
         isListed(_tokenAddress, _id)
         isSeller(_tokenAddress, _id)
     {
+        if(buyBackIfPriceIsLow(_tokenAddress, _id, _newPrice)) {
+            cancelListing(_tokenAddress, _id);
```

Comment:
Based on the code in both `buyBackIfPriceIsLow` (which burns the chara) and `cancelListing` (which tries to transfer the chara), I have a feeling the chara is always going to be burned before the cancellation tries transferring it, meaning it'll always error out in this case.